spec/posthog/feature_flag_spec.rb

```diff
@@ -1220,6 +1234,121 @@ class PostHog
+        # true because in ruby two equivalent datetimes return true even in a < comparison
+        expect(FeatureFlagsPoller.match_property(property_g, { 'key' => '2022-04-24 00:00:01' })).to be false
+
+        property_h = { 'key' => 'key', 'value' => '1m', 'operator' => 'is_relative_date_before'}
+        expect(FeatureFlagsPoller.match_property(property_h, { 'key' => '2022-03-01 00:00:00' })).to be true
+        expect(FeatureFlagsPoller.match_property(property_h, { 'key' => '2022-04-05 00:00:00' })).to be false
+
+        property_i = { 'key' => 'key', 'value' => '1y', 'operator' => 'is_relative_date_before'}
+        expect(FeatureFlagsPoller.match_property(property_i, { 'key' => '2021-04-28 00:00:00' })).to be true
+        expect(FeatureFlagsPoller.match_property(property_i, { 'key' => '2021-05-01 00:00:01' })).to be false
+
+        property_j = { 'key' => 'key', 'value' => '122h', 'operator' => 'is_relative_date_after'}
+        expect(FeatureFlagsPoller.match_property(property_j, { 'key' => '2022-05-01 00:00:00' })).to be true
+        expect(FeatureFlagsPoller.match_property(property_j, { 'key' => '2022-04-23 01:00:00' })).to be false
+
+        property_k = { 'key' => 'key', 'value' => '2d', 'operator' => 'is_relative_date_after'}
+        expect(FeatureFlagsPoller.match_property(property_k, { 'key' => '2022-05-01 00:00:00' })).to be true
+        expect(FeatureFlagsPoller.match_property(property_k, { 'key' => '2022-04-29 00:00:01' })).to be true
+        expect(FeatureFlagsPoller.match_property(property_k, { 'key' => '2022-04-29 00:00:00' })).to be false
+
+        property_l = { 'key' => 'key', 'value' => '02w', 'operator' => 'is_relative_date_after'}
+        expect(FeatureFlagsPoller.match_property(property_l, { 'key' => '2022-05-01 00:00:00' })).to be true
+        expect(FeatureFlagsPoller.match_property(property_l, { 'key' => '2022-04-16 00:00:00' })).to be false
+
+        property_m = { 'key' => 'key', 'value' => '1m', 'operator' => 'is_relative_date_after'}
+        expect(FeatureFlagsPoller.match_property(property_m, { 'key' => '2022-04-01 00:00:01' })).to be true
+        expect(FeatureFlagsPoller.match_property(property_m, { 'key' => '2022-04-01 00:00:00' })).to be false
+
+        property_n = { 'key' => 'key', 'value' => '1y', 'operator' => 'is_relative_date_after'}
+        expect(FeatureFlagsPoller.match_property(property_n, { 'key' => '2022-05-01 00:00:00' })).to be true
+        expect(FeatureFlagsPoller.match_property(property_n, { 'key' => '2021-05-01 00:00:01' })).to be true
+        expect(FeatureFlagsPoller.match_property(property_n, { 'key' => '2021-05-01 00:00:00' })).to be false
+        expect(FeatureFlagsPoller.match_property(property_n, { 'key' => '2021-04-30 00:00:00' })).to be false
+        expect(FeatureFlagsPoller.match_property(property_n, { 'key' => '2021-03-01 12:13:00' })).to be false
+
+      end
+    end
+
+    it 'nil property value with all operators' do
```

Comment:
Do we need to test this fully for ruby? Not sure it makes sense here since nil just returns empty when it's stringified. Unless we want to force nil values to be a string of nil.

Reply:
nope, makes sense, just make sure that operations don't crash with nil where we don't expect them to